fe/fe-core/src/main/java/org/apache/doris/datasource/mapping/DefaultIdentifierMapping.java

```diff
@@ -0,0 +1,268 @@
+
+        if (!conflictNames.isEmpty()) {
+            throw new RuntimeException(
+                    "Conflict table names found in remote database/schema: " + remoteDbName
+                            + " when lower_case_meta_names is true: " + conflictNames
+                            + ". Please set lower_case_meta_names to false or"
+                            + " use meta_name_mapping to specify the table names.");
+        }
+        return localTableNames;
+    }
+
+    @Override
+    public List<Column> fromRemoteColumnName(String remoteDatabaseName, String remoteTableName,
+            List<Column> remoteColumns) {
+        // If mapping is not required, return the original input
+        if (!isLowerCaseMetaNames && isMappingInvalid()) {
+            return remoteColumns;
+        }
+        JsonNode tablesNode = readAndParseJson(metaNamesMapping, "columns");
+
+        Map<String, String> columnNameMapping = Maps.newTreeMap();
+        if (tablesNode.isArray()) {
+            for (JsonNode node : tablesNode) {
+                String remoteDatabase = node.path("remoteDatabase").asText();
+                String remoteTable = node.path("remoteTable").asText();
+                if (remoteDatabaseName.equals(remoteDatabase) && remoteTable.equals(remoteTableName)) {
+                    String remoteColumn = node.path("remoteColumn").asText();
+                    String mapping = node.path("mapping").asText();
+                    columnNameMapping.put(remoteColumn, mapping);
+                }
+            }
+        }
+        localColumnToRemoteColumn.putIfAbsent(remoteDatabaseName, new ConcurrentHashMap<>());
+        localColumnToRemoteColumn.get(remoteDatabaseName).putIfAbsent(remoteTableName, new ConcurrentHashMap<>());
+
+        List<String> remoteColumnNames = Lists.newArrayList();
+        for (Column remoteColumn : remoteColumns) {
+            remoteColumnNames.add(remoteColumn.getName());
+        }
+
+        Map<String, List<String>> result = nameListToMapping(remoteColumnNames,
+                localColumnToRemoteColumn.get(remoteDatabaseName).get(remoteTableName),
+                columnNameMapping, isLowerCaseMetaNames);
+        List<String> localColumnNames = result.get("localNames");
+        List<String> conflictNames = result.get("conflictNames");
+        if (!conflictNames.isEmpty()) {
+            throw new RuntimeException(
+                    "Conflict column names found in remote database/schema: " + remoteDatabaseName
+                            + " in remote table: " + remoteTableName
+                            + " when lower_case_meta_names is true: " + conflictNames
+                            + ". Please set lower_case_meta_names to false or"
+                            + " use meta_name_mapping to specify the column names.");
+        }
+        for (int i = 0; i < remoteColumns.size(); i++) {
+            remoteColumns.get(i).setName(localColumnNames.get(i));
```

Comment:
And I think, the method `fromRemoteColumnName` should be act same as `fromRemoteDbName` and `fromRemoteTableName`, to return `List<String>` instead of `List<Column>`.
Which means these method only handle `Names`, you can wrap it and generate `Column` object outside